api/v1alpha1/linodeplacementgroup_types.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 Akamai Technologies, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// PlacementGroupFinalizer allows ReconcileLinodePG to clean up Linode resources associated
+	// with LinodePlacementGroup before removing it from the apiserver.
+	PlacementGroupFinalizer = "linodeplacementgroup.infrastructure.cluster.x-k8s.io"
+)
+
+// LinodePlacementGroupSpec defines the desired state of LinodePlacementGroup
+type LinodePlacementGroupSpec struct {
+	// +optional
+	PGID *int `json:"pgID,omitempty"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	Region string `json:"region"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:default=true
+	// +optional
+	IsStrict bool `json:"isStrict"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:default="anti_affinity:local"
+	// +kubebuilder:validation:Enum="anti_affinity:local"
+	// +optional
+	AffinityType string `json:"affinity_type"`
```

Comment:
This should probably be camel case like the other fields

Reply:
Also, we should probably do `+kubebuilder:validation:Enum="antiAffinity"`, dropping the ":local"

Reply:
The API expects it to be set to `anti_affinity:local` - I could switch up the enum - but that would mean more changes when making the translation from spec->createOptions

Reply:
want to make sure we separate the discussion on the enum vs field name. I agree we should camel case the json type here

Reply:
done - switched the json type